tests/check_docs.mts

```diff
@@ -0,0 +1,163 @@
+/**
+ * Require that documentation is kept up to date as the referenced files change.
+ *
+ * To do this, for each file in the docs directory, we look at the set of files
+ * listed in the front matter and check when each of them last changed. If any
+ * of them is more recent than either the last commit to the doc file or the
+ * commit listed in the "updated" field (whichever is more recent), then the doc
+ * is out of date.
+ */
+import child from "node:child_process";
+import { promises as fs } from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import { promisify } from "node:util";
+import matter from "gray-matter";
+
+const execFile = promisify(child.execFile);
+
+const filename = fileURLToPath(import.meta.url);
+const dirname = path.dirname(filename);
+
+interface NewerFile {
+  file: string;
+  commit: string;
+}
+
+type DocError =
+  | {
+      type: "out-of-date";
+      doc: string;
+      newerFiles: NewerFile[];
+    }
+  | {
+      type: "missing-front-matter";
+      doc: string;
+    }
+  | {
+      type: "missing-updated-field";
+      doc: string;
+    }
+  | {
+      type: "missing-files";
+      doc: string;
+    };
+
+const checkDoc = async (doc: string): Promise<DocError | undefined> => {
+  const docMatter = matter(await fs.readFile(doc));
+
+  if (!docMatter.data.updated) {
```

Comment:
We don't appear to use the value of `docMatter.data.updated` anywhere, besides verifying that it exists.  I guess it's really just a field to prompt the developer to touch the file in the commit (or a later commit in the branch), so the `rev-list` logic excludes it?

Reply:
Right. We could, like, verify that it is actually a timestamp if we wanted.

Reply:
Nah, I'm just verifying my understanding.

I think I'd rather leave it unchecked, to better handle the case there are two commits that update the same file in a day, which would make it hard to update the field if we verify that it's (purely) a date.

And if we made it a full date+time, that feels like it's asking each file for more specificity than is merited.  I think I'm fine leaving this as-is; just making sense of it.